platform/mellanox/docker-syncd-mlnx/ecmp_calculator/ecmp_calc.py

```diff
@@ -0,0 +1,491 @@
+                return port_channel
+        
+        raise KeyError("Failed to get port-channel name for interface {}".format(port))
+
+    def get_ingress_port_logical_idx(self):
+        for logical_index, sonic_port_name in self.ports_map.items():
+            if sonic_port_name == self.ingress_port:
+                return logical_index
+
+        raise KeyError("Failed to get logical index for interface {}".format(self.ingress_port))
+
+    # Get index in next hop array from which packet will egress 
+    def get_next_hop_index(self, ecmp_size):
+        logical = self.get_ingress_port_logical_idx()
+        
+        ecmp_hash = {
+            "ingress_port": str(hex(logical)),
+            "packet_info":self.packet['packet_info'],
+            "ecmp_size": ecmp_size,
+        }    
+            
+        self.debug_print("Calling hash calculator for ECMP")    
+        hash_result = self.call_hash_calculator({'ecmp_hash': ecmp_hash})
+        ecmp_hash_result = hash_result['ecmp_hash']
+        index = ecmp_hash_result['ecmp_index']  
+        
+        self.debug_print("Next hop index to which trafic will egress: {}".format(index))
+        return index  
+
+    # Get index in LAG memebrs array from which packet will egress
+    def get_lag_member_index(self, lag_size, flood_case = False):
+        logical = self.get_ingress_port_logical_idx()
+
+        lag_hash = {
+            "ingress_port": str(hex(logical)),
+            "packet_info": self.packet['packet_info'],
+            "lag_size": lag_size,
+        }
+    
+        self.debug_print("Calling hash calculator for LAG, flood case {}".format(True if flood_case else False))
+        hash_result = self.call_hash_calculator({"lag_hash": lag_hash})
+        lag_hash_result = hash_result["lag_hash"]
+        
+        if flood_case:
+            index = lag_hash_result['lag_mc_index']
+        else:
+            index = lag_hash_result['lag_index']
+
+        self.debug_print("Lag member index from which trafic will egress: {}".format(index))
+        return index
+        
+    # Get LAG memebr from which packet will egress
+    def get_lag_member(self, logical, flood_case = False):
+        lag_members = sx_lag_port_group_get(self.handle, logical)
+        lag_members.sort()
+        
+        member_index = self.get_lag_member_index(len(lag_members), flood_case)
+        lag_member = lag_members[member_index]
+        
+        self.debug_print("Lag member from which trafic will egress: {}".format(lag_member))
+        return lag_member
+        
+    def call_hash_calculator(self, input_dict):
+        with open(HASH_CALC_INPUT_FILE, "w") as outfile:
+            json.dump(input_dict, outfile)
+        
+        out = exec_cmd([HASH_CALC_PATH, '-c', HASH_CALC_INPUT_FILE, '-o', HASH_CALC_OUTPUT_FILE, '-d'])
+        self.debug_print ("Hash calculator output:\n{}".format(out))
+    
+        with open(HASH_CALC_OUTPUT_FILE, 'r') as openfile:
```

Comment:
you need to delete the file after using it. There is no delete for this file

Reply:
Temporary files deletion was added.